### PR TITLE
feat(space): allow nested objects in space

### LINKED ISSUE #351: Inconsistent property access
There's a *big* assumption in the `getValue` function that it operates on a scale (an array of primitive values), it's in line 168:

```js
return px(scale[n] || n)
```

That index/property-access means it'll only take a single-level object, which isn't an assumption i've found in other places. I also don't think it's particularly necessary, as you have a `get` function. This caused me a big headache, as for example `color` behaves differently. I think I can fix this with a backwards-compatible change (simply swapping it property access for `get`), would you be interested in that?

### PR DESCRIPTION
**Pull request by StevenLangbroek**
This uses the internal `get` helper to allow nested property access in `space`, rather than direct. This way space is ambivalent to the data structure of the theme, like other props are.

Are there other props that could use the same treatment?

Fixes #351

### CODE REVIEW

**Comment by StevenLangbroek** [comment]
Totally didn't realize this project doesn't have prettier, and I had it enabled in VS Code... Sorry! Want me to add a `.prettierrc`?

**Comment by jxnblk** [comment]
The actual change LGTM – if you want to add a `.prettierrc` file, I think this should be mostly inline with the current code style:

```json
{
  "semi": false,
  "singleQuote": true,
  "trailingComma": "es5"
}
```

**Comment by StevenLangbroek** [comment]
Added `.prettierrc` and `prettier` task to `package.json`. Only goes over `src` and `tests` folders, do you want `packages` as well? If you want I can add a precommit hook as well with `husky`.

**Comment by StevenLangbroek** [comment]
@jxnblk how can I help getting this merged?

**Comment by jxnblk** [comment]
Sorry it's the holidays and I've been away for a bit. Will review this again and hopefully merge soon

File: package.json [hunk 1]
```diff
@@ -7,6 +7,7 @@
   "sideEffects": false,
   "scripts": {
     "prepare": "npm run clean && npm run build:cjs && npm run build:esm",
+    "prettier": "prettier \"{src,test}/**/*.js\"",
```
Comment (jxnblk) [inline package.json]: I've used lint-staged for prettier in other projects. This is fine as-is but might be worth adding that as a follow up

Comment (StevenLangbroek) [reply package.json]: Yeah I'll open a new PR for that once this has been merged.

**Comment by StevenLangbroek** [comment]
Happy new year :)

**Comment by jxnblk** [comment]
Thanks for this! Should be published as v3.1.12 now 🎉

**Pull request merged**